### ISSUE TITLE
SemanticDB not found when using mill-bsp

### ISSUE THREAD

**Issue by asarkar** [scalameta/metals#5922]
I'm an IntelliJ defector due to the fact it refuses to recognize Scala 3 new syntax with Mill. My project is pretty basic, and the structure is shown below.

![1](https://github.com/scalameta/metals/assets/1302775/a58ed802-be89-4cae-9055-5f4527ca7bfe)

The `build.sc` is given below.

```
import mill._, scalalib._, scalafmt._

trait AdvancedScalaModule extends ScalaModule with ScalafmtModule {
  val baseDir = build.millSourcePath
  def scalaVersion = "3.3.1"
  def scalatestVersion = "3.2.17"
  def scalacheckVersion = "3.2.17.0"
  override def scalacOptions: T[Seq[String]] = Seq(
    "-encoding", "utf8",
    "-feature",
    "-Werror",
    "-explain",
    "-deprecation",
    "-unchecked",
    // Require then and do in control expressions
    "-new-syntax",
    "-rewrite",
    "-source", "future",
    // Tell Metals the location of SemanticDB
    // s"-Xplugin:semanticdb -sourceroot:$baseDir"
  )
}

object exercises extends AdvancedScalaModule {
  object test extends ScalaTests with TestModule.ScalaTest {
    // // use `::` for scala deps, `:` for java deps
    override def ivyDeps = Agg(
      ivy"org.scalactic::scalactic:$scalatestVersion",
      ivy"org.scalatest::scalatest:$scalatestVersion",
      ivy"org.scalatestplus::scalacheck-1-17:$scalacheckVersion"
    )
  }
}

object lectures extends AdvancedScalaModule
```

Notice the `-Xplugin:semanticdb -sourceroot:$baseDir` compiler option that is commented out. This is recommended by Metals Doctor (see screenshot below),  but when uncommented, produces an error in the Metals logs to the effect of "plugin not found".

This problem is documented in the [Mill documentation](https://mill-build.com/mill/Installation_IDE_Support.html#_build_server_protocol_bsp), but I couldn't find any open ticket corresponding to this problem.

> As of writing this, Metals 0.11.8 is the latest available version, which is not able to detect the fact, that Mill is already providing SemanticDB data. As a consequence, it reports missing SemanticDB data, although some SemanticDB-based features are already working.

### Expected behaviour:

No error in the logs or in the Metals Doctor report, and all functionality working as expected.

**Operating system:**
Mac OS X

**Java version:**
17.0.9

**Editor/extension:**
Visual Studio Code v1.84.2

**Metals version:**
1.1.0

### Extra context or search terms:
N/A.

### Workspace information:
 - **Scala versions:** 3.3.1; 2.13.12
 - **Build tools:** 0. mill
 - **Build servers:**
    0. mill-bsp v0.11.6
 - **All build tools in workspace:** Mill

![2](https://github.com/scalameta/metals/assets/1302775/914d1ff3-a00a-451e-87c9-64728935898c)

**Comment by tgodzik** [comment]
Thanks for reporting! The warning is a false positive one, you can ignore it. Next Metals release should not show it. Thanks to mill contributors, semanticdb is properly provided for Metals without the need to specify any additional options.

**Comment by asarkar** [comment]
@tgodzik So, you’re saying the note in the Mill docs is no longer correct? 

> As of writing this, Metals 0.11.8 is the latest available version, which is not able to detect the fact, that Mill is already providing SemanticDB data.

**Comment by asarkar** [comment]
> Tests do not currently work with Mill BSP unfortunately.

Is there a ticket for it? If not, I can open one?

Regarding Mill docs, I'll open a ticket on their gitHub Repo to update the docs.

**Comment by lefou** [comment]
> Tests do not currently work with Mill BSP unfortunately.

I think, this is because Metals is not using the dedicated `TestProvider` capability but the `DebugProvider` capability to run tests. I'm aware, that using the `DebugProvider` gives Metals more options in terms of selective test execution, but effectively for Mill it just means tests don't work. 

Also, we (Mill contributors) postponed the implementation of `DebugProvider` capability in Mill as IJ IDEA does not need it and Metals was supposed to also not need it eventually, as the plan was to start the DAP (debug adapter protocol) in Metals directly. But it seems to me, this no longer holds true, as I can't detect any work in that direction. @tgodzik Maybe, you could clarify the official Metals view point. Any outcomes is better that the status quo in which all parties (users as well as contributors to Mill and Metals) wait on each other.

**Comment by lefou** [comment]
> > Tests do not currently work with Mill BSP unfortunately.
> 
> Is there a ticket for it? If not, I can open one?

Not directly about test runners, but as Metals use the `DebugProvider` capability to run tests, this is the feature request in Mill: 

* https://github.com/com-lihaoyi/mill/issues/2225

**Comment by asarkar** [comment]
@lefou Although tangential to this ticket, IntelliJ didn’t recognize the Scala 3 new syntax (if-then, while-do etc.), so, the user experience with Mill using any of the two IDEs is really subpar. As a user, I just want to be able to get basic stuff done, without red underlines all over the code and tests not recognized.

**Comment by tgodzik** [comment]
> But it seems to me, this no longer holds true, as I can't detect any work in that direction. @tgodzik Maybe, you could clarify the official Metals view point. Any outcomes is better that the status quo in which all parties (users as well as contributors to Mill and Metals) wait on each other.

Yes, we do have a plan like that, but we were lacking time to do it. Let me open up an issue. I didn't concentrate on it, because you are able to do it via Bloop if you really needed it. However, it makes much more sense for it to be directly in Metals.

**Comment by tgodzik** [comment]
The exact issues will be fixed in the new release. We can focus on the moving scala-debug-adapter usage in MEtals in the other ticket
